src/foam/nanos/auth/ChangePasswordView.js

```diff
@@ -9,267 +9,166 @@ foam.CLASS({
       text-align: center;
+      padding-top: 1.5vh;
     }
   `,
 
-  sections: [
-    {
-      name: 'resetPasswordSection',
-      title: '',
-      help: 'Create a new password for your account',
-    }
-  ],
-
   properties: [
     {
       class: 'Boolean',
-      name: 'horizontal',
-      documentation: `This property toggles the view from a centered vertical view to a
-      horizontally property display view.`,
-      hidden: true
-    },
-    {
-      class: 'Boolean',
-      name: 'topBarShow',
+      name: 'showHeader',
       documentation: `This property toggles the view from having a top bar displayed.`,
       hidden: true,
-      value: true
-    },
-    {
-      class: 'String',
-      name: 'token',
-      documentation: `This property toggles the view from updating a user password to resetting a user password.`,
       factory: function() {
-        var searchParams = new URLSearchParams(location.search);
-        return searchParams.get('token');
-      },
-      hidden: true
+        // implement a logic to hide logo in certain places
+        return true;
+      }
     },
     {
-      class: 'Password',
-      name: 'originalPassword',
-      section: 'resetPasswordSection',
-      view: {
-        class: 'foam.u2.view.PasswordView',
-        passwordIcon: true
-      },
-      visibility: function(token) {
-        return ! token ? foam.u2.DisplayMode.RW : foam.u2.DisplayMode.HIDDEN;
+      class: 'Boolean',
+      name: 'isHorizontal',
+      documentation: 'setting this to true makes input fields in passwordModel to be displayed horizontally',
+      factory: function() {
+        return this.mode === 'update';
       }
     },
     {
-      class: 'Password',
-      name: 'newPassword',
-      section: 'resetPasswordSection',
-      view: {
-        class: 'foam.u2.view.PasswordView',
-        passwordIcon: true
-      },
-      minLength: 6
+      class: 'String',
+      name: 'mode',
+      documentation: 'name of password model used for this view',
+      factory: function() {
+        if ( window.location.hash.includes('personal-settings') ||
+             window.location.hash.includes('set-personal') ) {
+          return 'update';
+        }
+        const token = new URLSearchParams(location.search).get('token');
+        return token ? 'reset' : 'send';
+      }
     },
     {
-      class: 'Password',
-      name: 'confirmationPassword',
-      label: 'Confirm Password',
-      section: 'resetPasswordSection',
-      view: {
-        class: 'foam.u2.view.PasswordView',
-        passwordIcon: true
-      },
-      validationPredicates: [
-        {
-          args: ['newPassword', 'confirmationPassword'],
-          predicateFactory: function(e) {
-            return e.EQ(
-              foam.nanos.auth.ChangePasswordView.NEW_PASSWORD,
-              foam.nanos.auth.ChangePasswordView.CONFIRMATION_PASSWORD);
-          },
-          errorString: 'Passwords do not match.'
+      name: 'passwordModel',
```

Comment:
add a view to this property - { class: 'foam.u2.detail.VerticalDetailView' } should do it ... and get rid of the sectionView in initE. Also don't think we need to add anything to the SectionView and should be centering things as needed here.